src/code/z_bgcheck.c

```diff
@@ -1,432 +1,4713 @@
+    static Sphere16 sphere; // D_801EDBA8
+    static TriNorm tri;     // D_801EDBB0
+    Vec3f intersect;
+
+    CollisionPoly_GetVertices(poly, vtxList, tri.vtx);
+    CollisionPoly_GetNormalF(poly, &tri.plane.normal.x, &tri.plane.normal.y, &tri.plane.normal.z);
+    tri.plane.originDist = poly->dist;
+    sphere.center.x = center->x;
+    sphere.center.y = center->y;
+    sphere.center.z = center->z;
+    sphere.radius = radius;
+    return Math3D_ColSphereTri(&sphere, &tri, &intersect);
+}
+#else
+s32 CollisionPoly_SphVsPoly(CollisionPoly* poly, Vec3s* vtxList, Vec3f* center, f32 radius);
+#pragma GLOBAL_ASM("asm/non_matchings/code/z_bgcheck/CollisionPoly_SphVsPoly.s")
+#endif
+
+#ifdef NON_MATCHING
+/**
+ * Add poly to StaticLookup table
+ * Table is sorted by poly's smallest y vertex component
+ * `ssList` is the list to append a new poly to
+ * `polyList` is the CollisionPoly lookup list
+ * `vtxList` is the vertex lookup list
+ * `polyId` is the index of the poly in polyList to insert into the lookup table
+ */
+void StaticLookup_AddPolyToSSList(CollisionContext* colCtx, SSList* ssList, CollisionPoly* polyList, Vec3s* vtxList,
+                                  s16 polyId) {
+    SSNode* curNode;
+    SSNode* nextNode;
+    s32 polyYMin;
+    u16 newNodeId;
+    s16 curPolyId;
+
+    // if list is null
+    if (ssList->head == SS_NULL) {
+        SSNodeList_SetSSListHead(&colCtx->polyNodes, ssList, &polyId);
+        return;
+    }
+
+    polyYMin = CollisionPoly_GetMinY(&polyList[polyId], vtxList);
+
+    curNode = &colCtx->polyNodes.tbl[ssList->head];
+    curPolyId = curNode->polyId;
+
+    // if the poly being inserted has a lower y than the first poly
+    if (polyYMin < vtxList[COLPOLY_VTX_INDEX(polyList[curPolyId].flags_vIA)].y &&
+        polyYMin < vtxList[COLPOLY_VTX_INDEX(polyList[curPolyId].flags_vIB)].y &&
+        polyYMin < vtxList[polyList[curPolyId].vIC].y) {
+        SSNodeList_SetSSListHead(&colCtx->polyNodes, ssList, &polyId);
+        return;
+    }
+    while (true) {
+        // if at the end of the list
+        if (curNode->next == SS_NULL) {
+            newNodeId = SSNodeList_GetNextNodeIdx(&colCtx->polyNodes);
+            SSNode_SetValue(&colCtx->polyNodes.tbl[newNodeId], &polyId, SS_NULL);
+            curNode->next = newNodeId;
+            return;
+        }
+
+        nextNode = &colCtx->polyNodes.tbl[curNode->next];
+        curPolyId = nextNode->polyId;
+
+        // if the poly being inserted is lower than the next poly
+        if (polyYMin < vtxList[COLPOLY_VTX_INDEX(polyList[curPolyId].flags_vIA)].y &&
+            polyYMin < vtxList[COLPOLY_VTX_INDEX(polyList[curPolyId].flags_vIB)].y &&
+            polyYMin < vtxList[polyList[curPolyId].vIC].y) {
+            newNodeId = SSNodeList_GetNextNodeIdx(&colCtx->polyNodes);
+            SSNode_SetValue(&colCtx->polyNodes.tbl[newNodeId], &polyId, curNode->next);
+            curNode->next = newNodeId;
+            return;
```

Comment:
```suggestion
            break;
        }

        nextNode = &colCtx->polyNodes.tbl[curNode->next];
        curPolyId = nextNode->polyId;

        // if the poly being inserted is lower than the next poly
        if (polyYMin < vtxList[COLPOLY_VTX_INDEX(polyList[curPolyId].flags_vIA)].y &&
            polyYMin < vtxList[COLPOLY_VTX_INDEX(polyList[curPolyId].flags_vIB)].y &&
            polyYMin < vtxList[polyList[curPolyId].vIC].y) {
            newNodeId = SSNodeList_GetNextNodeIdx(&colCtx->polyNodes);
            SSNode_SetValue(&colCtx->polyNodes.tbl[newNodeId], &polyId, curNode->next);
            curNode->next = newNodeId;
            break;
```
Could you use breaks instead of returns here?